gRPC authentication proposal - using gRPC secure channels

JacekBartynowski opened the pull request:
<!---
Describe what your pull request does,
i.e. fix this bug and how, add a feature, fix documentation…
If you add a feature, please answer these questions:
- what do you use the feature for?
- how does labgrid benefit as a testing library from the feature?
- how did you verify the feature works?
- if hardware is needed for the feature, which hardware is supported and which
  hardware did you test with?
--->
**Description**

<!---
This checklist roughly outlines the steps for new features, remove and add tasks as needed:
--->
**Checklist**
- [x] Documentation for the feature
- [ ] Tests for the feature 
<!---
If you add a driver/resource or modify one:
--->
- [ ] The arguments and description in doc/configuration.rst have been updated
<!---
If you add a feature other drivers/resources can benefit from:
--->
- [x] Add a section on how to use the feature to doc/usage.rst
<!---
A library feature which other developers can use:
--->
- [x] Add a section on how to use the feature to doc/development.rst
<!---
Did you test the change locally? If yes, best to mention how you did it in the description section.
--->
- [x] PR has been tested
<!---
If your PR touched the man pages they have to be regenerated by calling make in the man subdirectory of the project
--->
- [x] Man pages have been regenerated

<!---
In case your PR fixes a bug, please reference it in the next line, i.e.
Fixes #[insert number without brackets here]
--->

Bastian-Krause committed:
driver/rawnetworkinterfacedriver: add interface up/down/wait_state

Add methods to the RawNetworkInterfaceDriver to set interfaces up and
down, as well as getting and waiting for the interface state.

This allows the driver to take more control over the interface,
preconfiguration is not needed anymore. Tests that expect the exporter
interface to be down (such as ethernet selftests, cable tests) are now
possible.

Note that the RawNetworkInterfaceDriver now brings the bound interface
up on activate and down on deactivate.

Signed-off-by: Bastian Krause <bst@pengutronix.de>
Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

Bastian-Krause committed:
driver/rawnetworkinterfacedriver: ethtool get/change EEE settings

Add interface Energy Efficient Ethernet (EEE) configuration
(`ethtool --set-eee`) support to the RawNetworkInterfaceDriver. This
allows configuring the EEE parameters eee, tx-lpi, tx-timer and advertise
on the bound interface.

Also add a `get_eee_settings()` method to query those settings. Note
that ethtool gained the required --json support for this sub command in
v6.10.

Signed-off-by: Bastian Krause <bst@pengutronix.de>
Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

Bastian-Krause committed:
driver/rawnetworkinterfacedriver: ethtool get/change pause settings

Add interface pause configuration (`ethtool --pause`) support to the
RawNetworkInterfaceDriver. This allows configuring the pause parameters
autoneg, rx and tx on the bound interface.

Also add a `get_pause_settings()` method to query those settings.

Signed-off-by: Bastian Krause <bst@pengutronix.de>
Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

Bastian-Krause committed:
driver/rawnetworkinterfacedriver: ethtool get/change settings

Add interface configuration (`ethtool --change`) support to the
RawNetworkInterfaceDriver. This allows configuring the bound interface
(speed, lanes, duplex, port, master-slave, mdix, autoneg, advertise,
phyad, xcvr, wol, sopass, msglvl).

Also add add a `get_settings()` method to query those settings. Note
that ethtool gained the required --json support for the default sub
command in v6.10.

Signed-off-by: Bastian Krause <bst@pengutronix.de>
Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

jacbar01-arm committed:
gRPC authentication proposal - using gRPC secure channels

Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

jacbar01-arm committed:
gRPC authentication proposal - using gRPC secure channels

Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

JacekBartynowski commented:
Here is a proposal for securing the gRPC channels (SSL encryption).
The authentication code will be reworked:
- possibly abstract interface would be added
- the concrete implementation would be done via additional package + certain 'entry points' as a way of the integration

ep1cman commented:
I imagine the private key (and thus the certificates?) would need to be generated on first start rather than included in the code, for anyone to see and copy?

JacekBartynowski commented:
These are self signed keys only for development and debugging purposes

Sent from Proton Mail Android

-------- Original Message --------
On 12/11/2024 11:49, Sebastian Goscik wrote:
I imagine the private key (and thus the certificates?) would need to be generated on first start rather than included in the code, for anyone to see and copy?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.￼Message ID: ***@***.***>

jacbar01-arm committed:
adding faciliation to use authentication plugins delivered as a separate Python packages - decoupling authentication code

Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

jacbar01-arm committed:
adding faciliation to use authentication plugins delivered as a separate Python packages - decoupling authentication code

jacbar01-arm committed:
gRPC authentication - updating the documentation

jacbar01-arm committed:
gRPC authentication - updating the documentation

Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

jacbar01-arm committed:
updating usage.rst description file

Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

jacbar01-arm committed:
updating usage.rst description file

jacbar01-arm committed:
gRPC documentation - minor updates

Signed-off-by: Jacek Bartynowski <jacek.bartynowski@arm.com>

jacbar01-arm committed:
gRPC documentation - minor updates

JacekBartynowski committed:
Merge branch 'master' into master

jacbar01-arm committed:
Merge branch 'master' of github.com:JacekBartynowski/labgrid

jacbar01-arm committed:
Merge branch 'master' of github.com:JacekBartynowski/labgrid

Signed-off-by: Jacek Bartynowski <Jacek.Bartynowski@arm.com>

jacbar01-arm committed:
Merge branch 'master' of github.com:JacekBartynowski/labgrid

jacbar01-arm committed:
gRPC authentication - fixing issues detected by ruff

Signed-off-by: Jacek Bartynowski <Jacek.Bartynowski@arm.com>

JacekBartynowski commented:
I update the Pull request, mainly added the instrumentation to select custom authentication plugins for the gRPC authentication.
These plugins would be delivered as a separate Python packages installed within the labgrid Python virtual environment and integrated/instantiated using Python _'entry-points'_ instrumentation.

jacbar01-arm committed:
gRPC authentication - adding instructions on SSL certificate and key generation

Signed-off-by: Jacek Bartynowski <Jacek.Bartynowski@arm.com>

JacekBartynowski commented:
I added instruction on generating SSL certificate and key + sample configuration file that could be re-used for this purpose. I updated the certificate + key files, they should contain now generic configuration.

Bastian-Krause commented:
See also #1885.
